internal/services/groups/groups_data_source.go

```diff
@@ -59,6 +59,12 @@ func groupsDataSource() *schema.Resource {
 				Optional:     true,
 				ExactlyOneOf: []string{"display_names", "object_ids", "return_all"},
 			},
+
+			"security_enabled": {
+				Description: "Retrieve only groups that are security_enabled groups",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
```

Comment:
This should also be computed since it will otherwise default to false, which would imply only non-security-enabled groups are returned by default.

```suggestion
				Computed: true,
			},
```